oarepo_requests/ui/theme/assets/semantic-ui/js/oarepo_requests_ui/common/components/TimelineRecordDiffSnapshotEvent.jsx

```diff
@@ -0,0 +1,222 @@
+  const [visibleAccordions, setVisibleAccordions] = useState({});
+
+  const eventIcon = getRequestStatusIcon("edited");
+  const feedMessage = getFeedMessage("edited");
+
+  const createdBy = event?.expanded?.created_by;
+  const creatorLabel =
+    createdBy?.profile?.full_name || createdBy?.username || createdBy?.email;
+
+  const toggleAccordion = (operationType) => {
+    setVisibleAccordions(prev => ({
+      ...prev,
+      [operationType]: !prev[operationType]
+    }));
+  };
+
+  // Parse and process diff data
+  const renderDiffTables = () => {
+    if (!event.payload?.diff) return null;
+
+    try {
+      const diffOperations = JSON.parse(event.payload.diff);
+
+      if (!_isArray(diffOperations)) {
+        return null;
+      }
+
+      // Handle case where there are no changes
+      if (diffOperations.length === 0) {
+        return (
+          <div>
+            <Icon name="info circle" />
+            {i18next.t("No changes detected between versions")}
+          </div>
+        );
+      }
+
+      // Group operations by type
+      const operationsByType = diffOperations.reduce((groups, op) => {
+        const operationType = op.op.toLowerCase();
+        if (!groups[operationType]) {
+          groups[operationType] = [];
+        }
+        groups[operationType].push(op);
+        return groups;
+      }, {});
+
+      // Format values for display
+      const formatValue = (value) => {
+        if (value === null || value === undefined) return "—";
+        if (_isArray(value) && _every(value, _isString)) return value.join(", ");
+        if (_isObjectLike(value))
+          return JSON.stringify(value, null, 2);
+        return String(value);
+      };
+
+      // Convert path to human readable format
+      const formatPath = (path) => {
+        return path
+          .replace(/^\//, "")
+          .replace(/\/(\d+)\//g, (match, arrayIndex) => {
+            return ` › ${parseInt(arrayIndex) + 1} › `;
+          })
+          .replace(/\/(\d+)$/, (match, arrayIndex) => {
+            return ` › ${parseInt(arrayIndex) + 1}`;
+          })
+          .replace(/\//g, " › ");
+      };
+
+      const renderOperationTable = (operations, operationType) => {
+        const operationConfig = {
+          add: {
+            title: i18next.t("Added"),
+            icon: "add",
+            color: "green"
+          },
+          remove: {
+            title: i18next.t("Removed"),
+            icon: "remove",
+            color: "red"
+          },
+          replace: {
+            title: i18next.t("Changed"),
+            icon: "pencil",
+            color: "orange"
+          }
+        };
+
+        const config = operationConfig[operationType];
+        if (!config || operations.length === 0) return null;
+
+        const isVisible = visibleAccordions[operationType] || false;
+
+        return (
+          <Accordion key={operationType} fluid className={`operation-accordion ${operationType}`}>
```

Comment:
This calls for extraction into separate sub-component :) It would simplify TimelineRecordDiffSnapshotEvent, so that it doesn't have to keep track of what accordions are open